addons/website_forum/models/forum_forum.py

```diff
@@ -156,6 +163,40 @@ def _compute_can_moderate(self):
         for forum in self:
             forum.can_moderate = self.env.user.karma >= forum.karma_moderate
 
+    @api.depends('post_ids', 'post_ids.tag_ids', 'post_ids.tag_ids.posts_count', 'tag_ids')
+    def _compute_tag_ids_usage(self):
+        forums_without_tags = self.filtered(lambda f: not f.tag_ids)
+        forums_without_tags.tag_most_used_ids = forums_without_tags.tag_unused_ids = False
+        forums_with_tags = self - forums_without_tags
+        if not forums_with_tags:
+            return
+
+        self.tag_ids.flush_recordset()
+        self.env.cr.execute(
```

Comment:
Any reason doing SQL and not using the ORM with a search + sort ? Preventive optimization or did you check on our prod ?

Reply:
Why would it be much slower using the ORM ? Isn't it possible to do a search fetch then a sort ?